Common/Product/SharedProject/IDEBuildLogger.cs

```diff
@@ -507,34 +503,30 @@ class NavigableErrorTask : ErrorTask {
         /// <summary>
         /// Sets the verbosity level.
         /// </summary>
-        private void SetVerbosity() {
-            if (!this.haveCachedVerbosity) {
-                this.Verbosity = LoggerVerbosity.Normal;
-
-                try {
-                    var settings = new ShellSettingsManager(serviceProvider);
-                    var store = settings.GetReadOnlySettingsStore(SettingsScope.UserSettings);
-                    if (store.CollectionExists(GeneralCollection) && store.PropertyExists(GeneralCollection, BuildVerbosityProperty)) {
-                        this.Verbosity = (LoggerVerbosity)store.GetInt32(GeneralCollection, BuildVerbosityProperty, (int)LoggerVerbosity.Normal);
-                    }
-                } catch (Exception ex) {
-                    var message = string.Format(
-                        "Unable to read verbosity option from the registry.{0}{1}",
-                        Environment.NewLine,
-                        ex.ToString()
-                    );
-                    this.QueueOutputText(MessageImportance.High, message);
+        public void RefreshVerbosity()
+        {
+            this.Verbosity = LoggerVerbosity.Normal;
+
+            try
+            {
+                serviceProvider.GetUIThread().MustBeCalledFromUIThread();
```

Comment:
OK, sounds good. Including the part about replacing with ThreadHelper later so it's a Release build runtime check too later on.